File: EduMS.Backend/src/EduMS.Application/M6_StatisticsReports/DTOs/ReportApprovals/ReportApprovalDto.cs
using System;

namespace EduMS.Application.M6_StatisticsReports.DTOs.ReportApprovals;

public class ReportApprovalDto
{
    public long Id { get; set; }
    public long SystemReportId { get; set; }
    public long SchoolId { get; set; }
    public DateTime SubmissionDate { get; set; } = DateTime.UtcNow;
    public long SubmittedByUserId { get; set; }
    public int ApprovalStatus { get; set; } = 1;
    public long? ReviewerId { get; set; }
    public DateTime? ReviewDate { get; set; }
    public string? Comments { get; set; }
    public string? RejectionReason { get; set; }
    public DateTime? ApprovalDate { get; set; }
    public long? ApprovedByUserId { get; set; }
    public string? DigitalSignatureHash { get; set; }
    public string? CertificateNumber { get; set; }
    public string? CertificatePath { get; set; }
    public bool IsFinal { get; set; }
    public string? Notes { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public long CreatedByUserId { get; set; }
    public DateTimeOffset? ModifiedAt { get; set; }
    public long? ModifiedByUserId { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAt { get; set; }
    public long? DeletedByUserId { get; set; }
    public Guid VersionToken { get; set; }
    public DateTimeOffset? LastSyncedAt { get; set; }
    public string SyncStatus { get; set; } = string.Empty;
}

File: 04_Frontend_Application/Edums-Frontend/src/app/modules/m6-statistics-reports/interfaces/report-approval.dto.ts
import { BaseAuditFields, ApprovalStatus } from './base.types';

export interface ReportApproval extends BaseAuditFields {
  systemReportId: number;
  schoolId: number;
  submissionDate: string;
  submittedByUserId: number;
  approvalStatus: ApprovalStatus;
  reviewerId: number | null;
  reviewDate: string | null;
  comments: string | null;
  rejectionReason: string | null;
  approvalDate: string | null;
  approvedByUserId: number | null;
  digitalSignatureHash: string | null;
  certificateNumber: string | null;
  certificatePath: string | null;
  isFinal: boolean;
  notes: string | null;
}

export type CreateReportApproval = Omit<ReportApproval, 'id' | 'createdAt' | 'modifiedAt'>;

export type UpdateReportApproval = Pick<ReportApproval, 'id'> & Partial<Omit<ReportApproval, 'id' | 'createdAt' | 'modifiedAt'>>;
